MacOS build script seems to stick to Python 3.7

Bug Report

==========
MacOS build script (most probably) relies on Python 3.7 in https://github.com/freeorion/freeorion/blob/master/CMakeLists.txt#L127


Environment
-----------
<!--
You don't need to provide the information in section when they are not useful in
the context of this issue.  For example a crashing hosted server may not need
information about the graphic card, but maybe depend on the operating system you
use.  When in doubt provide all information you have available.

You can find the FreeOrion version number in the lower right corner of the game
main screen.

When stating the operating system also note the version of the operating system
so, `Windows 8 Pro` instead of `Windows` or `Mac OSX Mountain Lion` instead of
`OSX`.
-->

* **FreeOrion Version**: master
* **Operating System**: macOS Ventura 13.1
* **Graphic card used**:
* **Fetched as** <!-- delete lines below that don't apply -->
  * Compiled from source


Description
-----------

The result of `cmake -G Xcode ../` is the following:
```
-- Found PythonInterp: /Users/Artem.Krosheninnikov/repositories/games/freeorion/build/dep/Frameworks/Python.framework/Versions/3.7/bin/python3.7 (found suitable version "3.7.12", minimum required is "3.7") 
-- Found PythonLibs: /Users/Artem.Krosheninnikov/repositories/games/freeorion/build/dep/Frameworks/Python.framework/Versions/3.7/lib/libpython3.7.dylib (found suitable version "3.7.12", minimum required is "3.7") 
-- Python library version detected 3.7.12
-- Boost python version 37
-- Boost 1.69.0 found.
-- Found Boost components:
   filesystem;iostreams;locale;log;log_setup;serialization;system;thread
-- Boost  found.
-- Found Boost components:
   python37
CMake Error at /usr/local/lib/cmake/Boost-1.81.0/BoostConfig.cmake:141 (find_package):
  Found package configuration file:

    /usr/local/lib/cmake/boost_python-1.81.0/boost_python-config.cmake

  but it set boost_python_FOUND to FALSE so package "boost_python" is
  considered to be NOT FOUND.  Reason given by package:

  No suitable build variant has been found.

  The following variants have been tried and rejected:

  * libboost_python311-mt.dylib (3.11, Boost_PYTHON_VERSION=3.7)

  * libboost_python311-mt.a (3.11, Boost_PYTHON_VERSION=3.7)

  * libboost_python311.dylib (3.11, Boost_PYTHON_VERSION=3.7)

  * libboost_python311.a (3.11, Boost_PYTHON_VERSION=3.7)

Call Stack (most recent call first):
  /usr/local/lib/cmake/Boost-1.81.0/BoostConfig.cmake:262 (boost_find_component)
  cmake/FindBoost.cmake:266 (find_package)
  CMakeLists.txt:284 (find_package)
```
<!--
Add a meaningful description of the bug you encountered.

If a screenshot or image helps to describe the issue content feel free to create
one.

Also attach the log files the game creates.  The log files The log files can be
found on
* Windows: %APPDATA%\FreeOrion
* MacOSX: ~/Library/"Application Support"/FreeOrion
* Linux: ${XDG_DATA_HOME:-~/.local/share}/freeorion
-->


Expected Result
---------------
<!--
When you enter an issue please add a description of what behaviour you would
expect from the game instead of the issue.
-->


Steps to reproduce
------------------
<!--
If the error only manifests itself after doing a certain number of actions
please add a list of steps to reproduce the bug.  If the error only occurs in
a certain game state please be sure to add a save game to the issue.

To upload save games please pack them into a Zip archive (*.zip) first.
-->
* First step.
* Second step.
* Another step.
* Issue occurs.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -125,6 +125,7 @@ if(APPLE)
     set(CMAKE_PREFIX_PATH "${CMAKE_BINARY_DIR}/dep/")
     set(CMAKE_FRAMEWORK_PATH "${CMAKE_BINARY_DIR}/dep/Frameworks")
     set(CMAKE_PROGRAM_PATH "${CMAKE_FRAMEWORK_PATH}/Python.framework/Versions/3.7/bin")
+    set(BOOST_ROOT "${CMAKE_BINARY_DIR}/dep/")
 endif()
 
 if(UNIX)

